route.php

```diff
@@ -62,18 +92,33 @@ public function decode($q)
         $args = preg_split('/[\/]/', $q);
 
         // get format (part of last argument after . i.e view.json)
-        $lastarg = sizeof($args) - 1;
-        $lastarg_split = preg_split('/[.]/', $args[$lastarg]);
-        if (count($lastarg_split) > 1) { $this->format = $lastarg_split[1]; }
-        $args[$lastarg] = $lastarg_split[0];
-
-        if (count($args) > 0) { $this->controller = $args[0]; }
-        if (count($args) > 1) { $this->action = $args[1]; }
-        if (count($args) > 2) { $this->subaction = $args[2]; }
-        
-        $this->method = "GET";
-        if ($_SERVER["REQUEST_METHOD"]=="POST") $this->method = "POST";
-        else if ($_SERVER["REQUEST_METHOD"]=="DELETE") $this->method = "DELETE";
-        else if ($_SERVER["REQUEST_METHOD"]=="PUT") $this->method = "PUT";
+        $lastArgIndex = sizeof($args) - 1;
+        $lastArgSplit = preg_split('/[.]/', $args[$lastArgIndex]);
+        if (count($lastArgSplit) > 1) {
+            $this->format = $lastArgSplit[1];
+        }
+        $args[$lastArgIndex] = $lastArgSplit[0];
+
+        if (count($args) > 0) {
+            $this->controller = $args[0];
+        }
+        if (count($args) > 1) {
+            $this->action = $args[1];
+        }
+        if (count($args) > 2) {
+            $this->subaction = $args[2];
+        }
+
+        if (in_array($requestMethod, ['POST', 'DELETE', 'PUT'])) {
+            $this->method = $requestMethod;
```

Comment:
You have to take inconsideration all modules that might exist that are not part of core repo.